src/ecs/resources/input/input_handler.rs

```diff
@@ -81,6 +28,8 @@ impl<'a> Iterator for Buttons<'a> {
 /// ```
 #[derive(Default)]
 pub struct InputHandler {
```

Comment:
@torkleyy ecs resources go on the heap anyways, the SmallVec just keeps us from having to do even more pointer dereferences.  If this makes the structure feel too bloated we can change it but I believe this approach would be more optimal.